src/Enclosure.php

```diff
@@ -251,10 +251,12 @@ public function __construct($link = null, $type = null, $length = null, $javascr
         $this->type = $type;
         $this->width = $width;
 
-        if (class_exists('idna_convert')) {
-            $idn = new \idna_convert();
+        if (function_exists('idn_to_ascii')) {
             $parsed = \SimplePie\Misc::parse_url($link);
-            $this->link = \SimplePie\Misc::compress_parse_url($parsed['scheme'], $idn->encode($parsed['authority']), $parsed['path'], $parsed['query'], $parsed['fragment']);
+            if ($parsed['authority'] !== '' && !ctype_print($parsed['authority'])) {
+                $authority = \idn_to_ascii($parsed['authority'], \IDNA_NONTRANSITIONAL_TO_ASCII, \INTL_IDNA_VARIANT_UTS46);
```

Comment:
It looks like the original code was already broken on authority strings containing non-ASCII characters in userinfo part of the URI, e.g. `čížek@háčkyčárky.cz`:

- This is what `idna_convert` gives: `xn--ek-mja1kox@xn--hkyrky-ptac70bc.cz`
- This is what Firefox converts: `%C4%8D%C3%AD%C5%BEek@xn--hkyrky-ptac70bc.cz`

It probably does not make sense to attempt to fix it when it will likely be corrected with switch to PSR-18.